script/fetchneedles

```diff
@@ -51,7 +52,9 @@ git_update() {
     branch="${1:-"master"}"
     git gc --auto --quiet
     git fetch -q origin
-    git rebase -q origin/"$branch"
+    # Clear any uncommitted changes that would prevent a rebase
+    [ "$force" = 1 ] && git reset --hard HEAD
+    git rebase -q origin/"$branch" || echo "Use force=1 to discard uncommitted changes before rebasing"
```

Comment:
Because of your echo call the script would not fail anymore in case of error but continue and likely fail later. Better also fail the script

Reply:
Silly me, forgot to add an `&& exit 1` there. Thank you for catching it!

I tested this again, with a dirty staging area and also wrong branch names for good measure (not sure if that's used in the wild, but the script supports it and should fail in that case).